src/absil/ilwritepdb.fs

```diff
@@ -0,0 +1,634 @@
+        use md5 = System.Security.Cryptography.MD5.Create()
+        let checkSum = md5.ComputeHash(file)
+        Some (guidSourceHashMD5, checkSum)
+    with _ -> None
+
+//------------------------------------------------------------------------------
+// PDB Writer.  The function [WritePdbInfo] abstracts the 
+// imperative calls to the Symbol Writer API.
+//------------------------------------------------------------------------------
+
+let sortMethods showTimes info =
+    reportTime showTimes (sprintf "PDB: Defined %d documents" info.Documents.Length)
+    Array.sortInPlaceBy (fun x -> x.MethToken) info.Methods
+    reportTime showTimes (sprintf "PDB: Sorted %d methods" info.Methods.Length)
+    ()
+
+let GetRowCounts tableRowCounts =
+    let builder = ImmutableArray.CreateBuilder<int>(tableRowCounts |> Array.length)
+    tableRowCounts |> Seq.iter(fun x -> builder.Add(x))
+    builder.MoveToImmutable()
+
+let fixupOverlappingSequencePoints fixupSPs showTimes methods =
+    // This next bit is a workaround.  The sequence points we get 
+    // from F# (which has nothing to do with this module) are actually expression 
+    // marks, i.e. the source ranges they denote are typically 
+    // nested, and each point indicates where the  
+    // code for an expression with a particular range begins.  
+    // This is in many ways a much more convenient form to emit. 
+    // However, it is not the form that debug tools accept nicely. 
+    // However, sequence points are really a non-overlapping, non-nested 
+    // partition of the source code of a method.  So here we shorten the 
+    // length of all sequence point marks so they do not go further than 
+    // the next sequence point in the source. 
+    let spCounts =  methods |> Array.map (fun x -> x.SequencePoints.Length)
+    let allSps = Array.concat (Array.map (fun x -> x.SequencePoints) methods |> Array.toList)
+    let allSps = Array.mapi (fun i sp -> (i,sp)) allSps
+    if fixupSPs then 
+        // sort the sequence points into source order 
+        Array.sortInPlaceWith (fun (_,sp1) (_,sp2) -> SequencePoint.orderBySource sp1 sp2) allSps
+        // shorten the ranges of any that overlap with following sequence points 
+        // sort the sequence points back into offset order 
+        for i = 0 to Array.length allSps - 2 do
+            let n,sp1 = allSps.[i]
+            let _,sp2 = allSps.[i+1]
+            if (sp1.Document = sp2.Document) && 
+               (sp1.EndLine > sp2.Line || 
+                (sp1.EndLine = sp2.Line &&
+                 sp1.EndColumn >= sp2.Column)) then
+              let adjustToPrevLine = (sp1.Line < sp2.Line)
+              allSps.[i] <-  n,{sp1 with EndLine = (if adjustToPrevLine then sp2.Line-1 else sp2.Line)
+                                         EndColumn = (if adjustToPrevLine then 80 else sp2.Column) }
+        reportTime showTimes (sprintf "PDB: fixupOverlappingSequencePoints %d" (allSps |> Array.length) )
```

Comment:
I think `reportTime` is existing, but if I'm correct we pay the cost of `sprintf` even if `showTimes` is `false`.

Should consider pass a function (a bit more noisy at call), I'd expect the func to be less overhead / allocations than sprintf which is not used most of the time.